test-app/tests/integration/components/bs-modal/footer-test.gts

```diff
@@ -134,31 +145,43 @@ module('Integration | Component | bs-modal/footer', function (hooks) {
   });
 
   test('submitDisabled disables submit button', async function (assert) {
-    this.set('disabled', true);
+    class State {
+      @tracked disabled = true;
+    }
+    const state = new State();
     await render(
-      hbs`<BsModal::Footer
-  @closeTitle='close'
-  @submitTitle='submit'
-  @submitDisabled={{this.disabled}}
-/>`,
+      <template>
+        <BsModalFooter
+          @closeTitle='close'
+          @submitTitle='submit'
+          @submitDisabled={{state.disabled}}
+        />
+      </template>,
     );
 
     assert
       .dom('.modal-footer button')
       .exists({ count: 2 }, 'Modal footer has two button.');
     assert.notOk(
```

Comment:
This is a great use case for the `isDisabled` DOM assertion. But could be also done in a follow-up.